pkg/kubelet/container/runtime.go

```diff
@@ -317,12 +317,17 @@ type ContainerStatus struct {
 // FindContainerStatusByName returns container status in the pod status with the given name.
 // When there are multiple containers' statuses with the same name, the first match will be returned.
 func (podStatus *PodStatus) FindContainerStatusByName(containerName string) *ContainerStatus {
+	var containerStatRet *ContainerStatus
 	for _, containerStatus := range podStatus.ContainerStatuses {
 		if containerStatus.Name == containerName {
-			return containerStatus
+			if containerStatRet == nil {
+				containerStatRet = containerStatus
+			} else if containerStatRet.State == ContainerStateExited {
+				containerStatRet = containerStatus
+			}
```

Comment:
I think you can use one `if ` here. Maybe it is like this
```
if containerStatRet == nil || containerStatus.State != ContainerStateExited {
				containerStatRet = containerStatus
			}
```

Reply:
The intention of else block is that non-exited container status would overwrite exited status.